app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -0,0 +1,495 @@
+                            if (str.contains("wbsetlabel")) {
+                                val descArr = str.split("|")
+                                if (descArr.size > 1) {
+                                    contributionLanguage = descArr[1]
+                                }
+                                editType = EDIT_TYPE_IMAGE_CAPTION
+                            } else if (str.contains("wbsetclaim")) {
+                                editType = EDIT_TYPE_IMAGE_TAG
+                            }
+                        }
+                        val con = Contribution("", userContribution.title, "", editType, "", DateUtil.iso8601DateParse(userContribution.timestamp),
+                                WikiSite.forLanguageCode(contributionLanguage), 0, if (editType == EDIT_TYPE_IMAGE_TAG) 1 else 0)
+                        if (!continuedImageContributions.add(con)) {
+                            continuedImageContributions.elementAt(continuedImageContributions.indexOf(con)).tagCount++
+                        }
+                    }
+                    Observable.fromIterable(continuedImageContributions)
+                }
+                .flatMap({ contribution: Contribution -> ServiceFactory.get(WikiSite(Service.COMMONS_URL)).getImageInfo(contribution.title, contribution.wikiSite.languageCode()) },
+                        { first: Contribution, second: MwQueryResponse -> Pair(first, second) })
+                .observeOn(AndroidSchedulers.mainThread())
+                .doAfterTerminate {
+                    for (continuedImageContribution in continuedImageContributions) {
+                        if (!imageContributions.add(continuedImageContribution)) {
+                            continuedImageContributions.elementAt(continuedImageContributions.indexOf(continuedImageContribution)).tagCount++
+                        }
+                    }
+                    createConsolidatedList()
+                }
+                .subscribe({ pair ->
+                    val response = pair.second
+                    val contribution = pair.first
+                    val page = response.query()!!.pages()!![0]
+                    if (page.imageInfo() != null) {
+                        val imageInfo = page.imageInfo()!!
+                        contribution.description = imageInfo.metadata!!.imageDescription()
+                        contribution.imageUrl = imageInfo.originalUrl
+                        contribution.title = contribution.title.replace("File:", "")
+                        if (contribution.editType == EDIT_TYPE_IMAGE_TAG) {
+                            if (!page.imageLabels.isNullOrEmpty()) {
+                                var labelString = ""
+                                for (imageLabel in page.imageLabels) {
+                                    labelString = imageLabel.label + "," + labelString
+                                }
+                                contribution.description = labelString
```

Comment:
It can be simplified as `contribution.description = page.imageLabels.joinToString(separator = ",")`